src/units/abilities.cpp

```diff
@@ -182,34 +182,52 @@ bool affects_side(const config& cfg, std::size_t side, std::size_t other_side)
 		return cfg["affect_allies"].to_bool();
 }
 
+	// This function defines in which direction loc is relative to from_loc
+	// either by pointing to the hexagon loc is on or by pointing to the hexagon adjacent to from_loc closest to loc.
+	utils::optional<int> find_direction(const map_location& loc, const map_location& from_loc, unsigned int distance)
+	{
+		const auto adjacent = get_adjacent_tiles(from_loc);
+		for(unsigned j = 0; j < adjacent.size(); ++j) {
+			bool adj_or_dist = distance != 1 ? distance_between(adjacent[j], loc) == (distance - 1) : adjacent[j] == loc;
+			if(adj_or_dist) {
+				return j;
+			}
+		}
+		return utils::nullopt;
+	}
 }
 
 bool unit::get_ability_bool(const std::string& tag_name, const map_location& loc) const
 {
+	// Check that the unit has an ability of tag_name type which meets the conditions to be active.
+	// If so, return true.
 	for (const config &i : this->abilities_.child_range(tag_name)) {
 		if (get_self_ability_bool(i, tag_name, loc))
 		{
 			return true;
 		}
 	}
 
+	// If the unit does not have abilities that match the criteria, check if adjacent units or elsewhere on the map have active abilities
+	// with the [affect_adjacent] subtag that could affect the unit.
 	const unit_map& units = get_unit_map();
 
-	const auto adjacent = get_adjacent_tiles(loc);
-	for(unsigned i = 0; i < adjacent.size(); ++i) {
-		const unit_map::const_iterator it = units.find(adjacent[i]);
-		if (it == units.end() || it->incapacitated())
+	// Check for each unit present on the map that it corresponds to the criteria
+	// (possession of an ability with [affect_adjacent] via a bollean variable, not incapacitated,
+	// different from the central unit, that the ability is of the right type, detailed verification of each ability),
+	// if so return true.
+	for(const unit& unit_itor : units) {
```

Comment:
These are not iterators since you're using a range for loop. Please use another variable name (`unit` or just `u`) here and anywhere else you have `unit_itor`.

Reply:
replaced by 'u' in abilities.cpp and by 'unit' in other files.